beps/tool-persist/1406_support_for_individual_init_parameters_in_persist_client.md

```diff
@@ -0,0 +1,184 @@
+# Support for Individual Init Parameters in Persist Clients
+
+- Authors - @TharmiganK
+- Reviewed by - @daneshk
+- Created date - 2025-12-05
+- Updated date - 2025-12-05
+- Issue - [#1406](https://github.com/ballerina-platform/ballerina-spec/issues/1406)
+- State - Submitted
+
+## Summary
+
+This proposal introduces support for individual initialization parameters in Ballerina persist clients as an alternative to the existing module-level configurable-based approach. This enables a more low-code friendly and programmatic way to initialize database connections.
+
+## Motivation
+
+Currently, Ballerina persist clients are initialized using configurables bound to module-level variables and exposed through a parameter-less `init()` function:
+
+```ballerina
+// Generated configurables (module-level)
+configurable string host = ?;
+configurable int port = ?;
+configurable string user = ?;
+configurable string password = ?;
+configurable string database = ?;
+
+// Generated init function
+public function init() returns Client|error {
+    final Client dbClient = check new (); // Uses configurables internally
+    return new (dbClient);
+}
+```
+
+This approach presents significant challenges:
+
+1. **Client reusability**: The parameter-less `init()` function makes it impossible to initialize the same client module with different configurations. Applications requiring multiple database connections (e.g., read replicas, multi-tenant scenarios) cannot reuse the generated client code and must resort to workarounds.
+
+2. **Low-code developer experience**: In low-code environments, the generated configurables appear as separate, unrelated variables. The association between these variables and the database client is hidden within the generated code, making it unclear to developers which parameters are required for client initialization. This disconnect significantly degrades the developer experience and makes the client initialization opaque.
+
+This proposal introduces support for individual initialization parameters, enabling explicit, self-documenting client initialization while maintaining backward compatibility with the existing configurable-based approach.
+
+## Goals
+
+- Enable persist clients to be initialized with individual parameters directly in code
+- Support both configurable-based and parameter-based initialization approaches
+- Maintain backward compatibility with existing persist applications
+- Apply this feature to all SQL-based datastores (MySQL, PostgreSQL, MSSQL, H2)
```

Comment:
What happens when a user tries with a flag with non-SQL datastores like  Redis DB. Shall we include that here